### PR TITLE
Introduce before_feature and after_feature to the framework

### PR DESCRIPTION
**Pull request by ypu**
This pullreq aimed to summarize the common setup and cleanup functions for each features and make it separated from the real test steps. This will help to reduce the duplicate code in features.

And also have to make it easier to switch the target host between dynamic and static.

### CODE REVIEW

File: environment.py [hunk 1]
```diff
@@ -168,3 +179,45 @@ def after_step(context, step):
             file_logger.info('Ansible Output: %s' % context.result)
             print('Ansible Output: %s' % context.result)
 
+
+def before_feature(context, feature):
```
Comment (miabbott) [inline environment.py:183]: I'd like to see a little bit of comments here explaining what this function is trying to do.  I managed to figure it out, but some description would be helpful.

File: environment.py [hunk 2]
```diff
@@ -168,3 +179,45 @@ def after_step(context, step):
...
+        func = getattr(env_setup, test_tag)
+        try:
+            func(context)
+        except Exception, err:
+            print("%s failed with following error: %s" % (test_tag,
+                                                          err.message))
+
+
+
+def after_feature(context, feature):
```
Comment (miabbott) [inline environment.py:213]: Same comment for `before_feature()` applies here.

**Comment by miabbott** [comment]
@ypu Thanks for the PR.  I made two comments to improve the documentation, but otherwise LGTM.

**Comment by chuanchang** [comment]
I like this feature, it's what I want, it will let our code more concise, so ACK for me.

**Comment by ypu** [comment]
@miabbott Thanks for your advice. And add some docstring for before_feature and after_feature. Can you help to review it again?

**Comment by miabbott** [comment]
Thanks @ypu!

**Pull request merged**